gslb/ingestion/gslb_host_rule.go

```diff
@@ -186,6 +187,98 @@ func isGslbPoolAlgorithmValid(algoSettings *gslbhralphav1.PoolAlgorithmSettings)
 	}
 }
 
+func getHMFromName(name string, gdp bool) (*models.HealthMonitor, error) {
+	aviClient := avictrl.SharedAviClients().AviClient[0]
+	uri := "api/healthmonitor?name=" + name
+
+	result, err := gslbutils.GetUriFromAvi(uri, aviClient, gdp)
+	if err != nil {
+		gslbutils.Errf("Error in getting uri %s from Avi: %v", uri, err)
+		return nil, err
+	}
+	if result.Count == 0 {
+		gslbutils.Errf("Health Monitor %s does not exist", name)
+		return nil, fmt.Errorf("health Monitor %s does not exist", name)
+	}
+	gslbutils.Logf("health monitor %s fetched from controller", name)
+	elems := make([]json.RawMessage, result.Count)
+	err = json.Unmarshal(result.Results, &elems)
+	if err != nil {
+		gslbutils.Errf("failed to unmarshal health monitor data for ref %s: %v", name, err)
+		return nil, err
+	}
+	hm := models.HealthMonitor{}
+	err = json.Unmarshal(elems[0], &hm)
+	if err != nil {
+		gslbutils.Errf("failed to unmarshal the first health monitor element: %v", err)
+		return nil, err
+	}
+	return &hm, nil
+}
+
+func isHealthMonitorTemplatePresentInCache(hmName string) bool {
+	aviHmCache := avictrl.GetAviHmCache()
+
+	obj, present := aviHmCache.AviHmCacheGet(avictrl.TenantName{Tenant: utils.ADMIN_NS, Name: hmName})
+	if !present {
+		return false
+	}
+	hmObj, ok := obj.(*avictrl.AviHmObj)
+	return ok && hmObj.CustomHmSettings != nil
+}
+
+// Checks whether the template is present in the controller. If present, it adds the contents of the template to the cache which will
+// used for the creation of health monitors. If the template is not present in the controller, the GDP/GSLBHostRule
+// will be rejected.
+func validateAndAddHmTemplateToCache(hmTemplate string, gdp bool, fullSync bool) error {
```

Comment:
Can we move this function to a common location like `ingestion/gslb.go` as both `gdp_controller.go` and `gslb_host_rule.go` are using it.

Reply:
I have moved the code to a common place.